packages/hoprd/crates/hopr-lib/src/chain.rs

```diff
@@ -322,32 +335,72 @@ impl ProtocolConfig {
     }
 }
 
+type ActiveTxExecutor = EthereumTransactionExecutor<
+    TypedTransaction,
+    RpcEthereumClient<RpcOperations<JsonRpcClient>>,
+    SafePayloadGenerator,
+>;
+
 #[cfg(feature = "wasm")]
-pub fn build_chain_components<Db, TxExec: TransactionExecutor + 'static>(
-    me: Address,
+pub fn build_chain_components<Db>(
+    me_onchain: &ChainKeypair,
+    chain_config: ChainNetworkConfig,
+    contract_addrs: ContractAddresses,
+    module_address: Address,
     db: Arc<RwLock<Db>>,
-    tx_executor: TxExec,
-) -> (TransactionQueue<Db>, CoreEthereumActions<Db>)
+) -> (
+    ActionQueue<Db, IndexerActionTracker, ActiveTxExecutor>,
+    CoreEthereumActions<Db>,
+    RpcOperations<JsonRpcClient>,
+)
 where
     Db: HoprCoreEthereumDbActions + Clone + 'static,
 {
-    let tx_queue = TransactionQueue::new(db.clone(), Box::new(tx_executor));
+    let rpc_client = build_json_rpc_client(
+        &chain_config.chain.default_provider, // TODO: is this the right value ?
```

Comment:
I think so, links to the hopr default and configurable value from the yaml/cmd line args.